src/tests/test_finite_difference.py

```diff
@@ -1,64 +1,60 @@
 from __future__ import absolute_import
 from builtins import range
-from __init__ import ForceBalanceTestCase
-import unittest
 import numpy
 import forcebalance
 import re, sys
 from math import cos, sin, pi
+import pytest
+# class TestFiniteDifference(ForceBalanceTestCase):
+# class TestFiniteDifference:
+#     # def setUp(self):
+#     @classmethod
+# def setup_class(cls):
+    # functions is a list of 3-tuples containing (function p, d/dp, d^2/dp^2)
+functions = []
 
-class TestFiniteDifference(ForceBalanceTestCase):
-    def setUp(self):
-        # functions is a list of 3-tuples containing (function p, d/dp, d^2/dp^2)
-        self.functions = []
+# f(x) = 2x
```

Comment:
Can we continue to have things organized in classes? Do we lose anything essential by removing ForceBalanceTestCase? Can we replace it with equivalent functionality from pytest?

Reply:
We can keep things organized in classes. See the changes in `test_tinkerio` and `__init__`.

Reply:
I see this comment was made after the others, so I'm interested in knowing how you keep things organized in classes.  I do see value in it (from comments above) but it's worth discussing whether the value is worth the extra "bureaucracy".

Reply:
I'm pretty new to test structure as well. Generally I just use classes for bookkeeping -- The use of inheritance from `ForceBalanceTestCase` is interesting, and is probably worth keeping around. I'm less certain about `ForceBalanceTestResult` and `ForceBalanceTestRunner` -- I haven't read too closely, but the functionality there may be redundant with what's built in to pytest.

The "bureaucracy" goes both ways. There have been times where the act of grouping tests has made me find redundancies or gaps in coverage around certain functionalities. The FB tests seem very well thought-out, so there are probably fewer of that type of inconsistency. I could go either way on the use of classes, and removing them is as simple as changing `setup_class` to `setup_module` and `setup_method` to `setup_function`.